core/src/main/java/com/sequenceiq/cloudbreak/service/clusterdefinition/ClusterDefinitionService.java

```diff
@@ -150,6 +151,40 @@ public ClusterDefinition create(Workspace workspace, ClusterDefinition clusterDe
         return clusterDefinitions;
     }
 
+    public ClusterDefinition getByNameForWorkspaceAndLoadDefaultsIfNecessary(String name, Workspace workspace) {
+        Set<ClusterDefinition> clusterDefinitions = clusterDefinitionRepository.findAllByNotDeletedInWorkspace(workspace.getId());
```

Comment:
Can we select the ones that are default by default rather than fetching all?

Reply:
I think not, because if you have a custom template with custom definition then we won't find it